File: C++/Lab_9/Lab_9.cpp
#include <string>
#include <iostream>

using namespace std;
bool palindrome(string);
int function(string);

int main()
{
	string myString;
	cout << "Enter your sentence" << endl;
	getline(cin, myString);
	// введення рядка

	int numOfPalindromes = function(myString);
	cout << "Total number of palindromes: " << numOfPalindromes;
	// виведення результату
}

int function(string myString)
{
	int gap = myString.find(" ", 0);
	int number = 0;
	// знаходження позиції першого пробіла
	if (gap == -1 && !(myString.empty()))
		gap = 0;
	// перевірка на випадок, якщо користувач введе тільки одне слово
	while (gap != -1)
	{
		gap = myString.find(" ", 0);
		string word = myString.substr(0, gap);
		// виділення першого слова
		if (palindrome(word)) {
			number += 1;
			myString.erase(0, word.length() + 1);
			// видалення обробленого слова
		}
		else {
			myString.erase(0, word.length() + 1);
		}

	}
	return number;
}

bool palindrome(string word)
// функція для визначення чи є слово паліндромом
{
	string reverseWord;
	for (int i = word.length() - 1; i >= 0; i--)
		reverseWord += word[i];
	// цикл який перевертає слово
	if (word == reverseWord && !(reverseWord.empty()))
	// додаткова перевірка, щоб уникнути сприйняття пробілів як паліндромів
	{
		cout << "Word " << word << " is palindrome\n";
		return true;
	}
	else
		return false;
}

File: Python/Lab_9/Lab_9.py
def palindrome(word):
    if not word:
        return False
    # додаткова перевірка на пустий рядок
    reversed_word = word[::-1]
    # перевертання слова за допомогою зрізу
    return reversed_word == word

def function(my_string):
    my_string = my_string.split(" ")
    # розділення всіх слів і запис їх у список
    num_of_words = len(my_string)
    number = 0
    for i in range (num_of_words):
    # цикл для перевірки кожного слова
        if palindrome(my_string[i]):
        # перевирка чи є слово паліндромом
            print("Word ", my_string[i], " is palindrome")
            number += 1
    return number

my_string = input("Enter your sentence\n")
# введення рядка
num_of_palindromes = function(my_string)
# визначення кількості паліндромів
print("Total number of palindromes is ", num_of_palindromes)
